plugins/InspectionGadgets/testsrc/com/siyeh/ig/fixes/controlflow/ExpressionMayBeFactorizedFixTest.java

```diff
@@ -174,6 +171,32 @@ public void testDoNotCleanupWithXOrOperator() {
                                "}\n");
   }
 
+  public void testDoNotCleanupWithFinalLazyActiveExpression() {
+    assertQuickfixNotAvailable(InspectionGadgetsBundle.message("if.may.be.factorized.quickfix"),
+                               "class X {\n" +
+                               "  boolean test() {\n" +
+                               "    boolean duplicateExpression = true;\n" +
+                               "    boolean thenExpression = true;\n" +
+                               "    boolean updatedExpression = true;\n" +
+                               "    boolean newValue = false;\n" +
+                               "    boolean evaluation = (duplicateExpression || thenExpression) &&/**/ (duplicateExpression || updatedExpression = newValue);\n" +
+                               "    return updatedExpression;\n" +
+                               "  }\n" +
+                               "}\n");
+  }
+
+  public void testDoNotCleanupWithNotFinalEagerActiveExpression() {
+    assertQuickfixNotAvailable(InspectionGadgetsBundle.message("if.may.be.factorized.quickfix"),
+                               "class X {\n" +
+                               "  boolean test() {\n" +
+                               "    boolean duplicateExpression = true;\n" +
+                               "    boolean newValue = false;\n" +
+                               "    boolean elseExpression = false;\n" +
+                               "    return (duplicateExpression | duplicateExpression = newValue) &/**/ (duplicateExpression | elseExpression);\n" +
```

Comment:
This test case is red code (an expression instead of a variable on the left side in the assignment expression), is it meant to be?

Reply:
It wasn't intentional. I fixed it.